src/test/java/com/thealgorithms/strings/LetterCombinationsOfPhoneNumberTest.java

```diff
@@ -1,45 +1,24 @@
 
 import java.util.Arrays;
 import java.util.List;
-import org.junit.jupiter.api.Test;
+import java.util.stream.Stream;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
 
 public class LetterCombinationsOfPhoneNumberTest {
 
-    @Test
-    public void letterCombinationsOfPhoneNumber() {
-        LetterCombinationsOfPhoneNumber.generateNumberToCharMap();
-
-        // ** Test 1 **
-        // Input: digits = ""
-        // Output: []
-        int[] numbers1 = {};
-        List<String> output1 = Arrays.asList("");
-        assertTrue(LetterCombinationsOfPhoneNumber.printWords(numbers1, numbers1.length, 0, "").equals(output1));
-
-        // ** Test 2 **
-        // Input: digits = "2"
-        // Output: ["a","b","c"]
-        int[] numbers2 = {2};
-        List<String> output2 = Arrays.asList("a", "b", "c");
-        assertTrue(LetterCombinationsOfPhoneNumber.printWords(numbers2, numbers2.length, 0, "").equals(output2));
-
-        // ** Test 3 **
-        // Input: digits = "23"
-        // Output: ["ad","ae","af","bd","be","bf","cd","ce","cf"]
-        int[] numbers3 = {2, 3};
-        List<String> output3 = Arrays.asList("ad", "ae", "af", "bd", "be", "bf", "cd", "ce", "cf");
-        assertTrue(LetterCombinationsOfPhoneNumber.printWords(numbers3, numbers3.length, 0, "").equals(output3));
+    @ParameterizedTest
+    @MethodSource("provideTestCases")
+    public void testLetterCombinationsOfPhoneNumber(int[] numbers, List<String> expectedOutput) {
+        assertEquals(expectedOutput, LetterCombinationsOfPhoneNumber.getCombinations(numbers));
+    }
 
-        // ** Test 4 **
-        // Input: digits = "234"
-        // Output: ["adg", "adh", "adi", "aeg", "aeh", "aei", "afg", "afh", "afi",
-        // "bdg", "bdh", "bdi", "beg", "beh", "bei", "bfg", "bfh", "bfi", "cdg", "cdh",
-        // "cdi", "ceg", "ceh", "cei", "cfg", "cfh", "cfi"]
-        int[] numbers4 = {2, 3, 4};
-        List<String> output4 = Arrays.asList("adg", "adh", "adi", "aeg", "aeh", "aei", "afg", "afh", "afi", "bdg", "bdh", "bdi", "beg", "beh", "bei", "bfg", "bfh", "bfi", "cdg", "cdh", "cdi", "ceg", "ceh", "cei", "cfg", "cfh", "cfi");
-        assertTrue(LetterCombinationsOfPhoneNumber.printWords(numbers4, numbers4.length, 0, "").equals(output4));
+    private static Stream<Arguments> provideTestCases() {
+        return Stream.of(Arguments.of(null, List.of("")), Arguments.of(new int[] {}, List.of("")), Arguments.of(new int[] {2}, Arrays.asList("a", "b", "c")), Arguments.of(new int[] {2, 3}, Arrays.asList("ad", "ae", "af", "bd", "be", "bf", "cd", "ce", "cf")),
+            Arguments.of(new int[] {2, 3, 4}, Arrays.asList("adg", "adh", "adi", "aeg", "aeh", "aei", "afg", "afh", "afi", "bdg", "bdh", "bdi", "beg", "beh", "bei", "bfg", "bfh", "bfi", "cdg", "cdh", "cdi", "ceg", "ceh", "cei", "cfg", "cfh", "cfi")));
```

Comment:
Please add a test case with `{8, 4}` (_not sorted input_) and `{3, 3}` (_duplicates_).